src/main/java/it/pagopa/interop/probing/eservice/operations/service/EserviceServiceImpl.java

```diff
@@ -46,25 +45,23 @@ public Long saveEservice(SaveEserviceDto inputData) {
     UUID eserviceId = UUID.fromString(inputData.getEserviceId());
     UUID versionId = UUID.fromString(inputData.getVersionId());
     Eservice eServiceToUpdate =
-        eserviceRepository.findByEserviceIdAndVersionId(eserviceId, versionId).orElse(null);
+        eserviceRepository.findByEserviceIdAndVersionId(eserviceId, versionId).orElseGet(
+            () -> Eservice.builder().versionId(versionId).eserviceId(eserviceId).lockVersion(1)
+                .versionNumber(Integer.valueOf(inputData.getVersionNumber())).build());
 
-    if (Objects.isNull(eServiceToUpdate)) {
-      eServiceToUpdate = Eservice.builder().versionId(versionId).eserviceId(eserviceId)
-          .lockVersion(1).versionNumber(Integer.valueOf(inputData.getVersionNumber())).build();
-    }
-
-    eServiceToUpdate.setEserviceName(inputData.getName());
-    eServiceToUpdate.setProducerName(inputData.getProducerName());
-    eServiceToUpdate.setBasePath(inputData.getBasePath());
-    eServiceToUpdate.setTechnology(inputData.getTechnology());
-    eServiceToUpdate.setState(inputData.getState());
+    eServiceToUpdate.setEserviceName(inputData.getName())
+        .setProducerName(inputData.getProducerName()).setBasePath(inputData.getBasePath())
```

Comment:
If you use fluent = true (chain and fluent are true by default), you can omit set and get and it's more readable